scanpy/tools/_leiden.py

```diff
@@ -141,18 +141,43 @@ def leiden(
         categories=natsorted(np.unique(groups).astype('U')),
     )
     # store information on the clustering parameters
-    adata.uns['leiden'] = {}
-    adata.uns['leiden']['params'] = dict(
+    uns_key = key_added if key_added.startswith('leiden') else f'leiden_{key_added}'
+    adata.uns[uns_key] = {}
+    adata.uns[uns_key]['params'] = dict(
         resolution=resolution,
         random_state=random_state,
         n_iterations=n_iterations,
+        use_weights=use_weights,
+        directed=directed,
+        partition_type=None if partition_type is None else partition_type.__name__,
     )
+    if 'quality' in dir(part):
```

Comment:
Could this be:

```suggestion
   if isinstance(part, MutableVertexPartition):
```

Reply:
This part is gone now, I simply report standard modularity regardless of the partititon_type.